### PR TITLE
add support for .env configuration

### LINKED ISSUE #1064: Add support for loading configuration from the environment
`pc` should look for its configuration in env vars and use those before using the `pcconfig.py` file or the defaults from `pynecone/constants.py`. Adding in support for using env vars for feeding configuration into pynecone will make using containers easier. 

I would prefix the configuration vars in  `pynecone/constants.py` that make sense for running in production with `PYNECONE_`  and if that is set in the environment, use that over what is set in the `pcconfig.py` file. 

For example I want to be able to set my `API_URL` as an environment variable when running Pynecone inside a container. If the above is implemented, we can then set `PYNECONE_API_URL=api.mycoolsite.biz` at container runtime and we can then use Traefik, Caddy, or Nginx to proxy requests to the container.

### PR DESCRIPTION
**Pull request by ElijahAhianyo**
In this PR you can add a .env file to your config class. 

```python


class ExpConfig(pc.Config):
    env_path = "envs/.env"

config = ExpConfig(
    app_name="exp",
    db_url="sqlite:///pynecone.db",
    env=pc.Env.DEV,
)
```
or as key work argument:

```python

class ExpConfig(pc.Config):
    pass

config = ExpConfig(
    app_name="exp",
    db_url="sqlite:///pynecone.db",
    env=pc.Env.DEV,
    env_path = "envs/.env"
)
```
By default, pynecone looks for a .env file($PROJECT_ROOT/.env) in your root directory. So if you dont specify the path but provide a .env file, it will be loaded into the os env. 
Also by default, the .env file overides os env variables. In order of precedence .env >> commandline args >> os env >> config file. However, to change this behavior/order, set `override_os_envs` to `False`.  Setting `override_os_envs=False` changes the precedence order to command line args >> os env >> .env file >> config file.

This PR also contains a breaking change: `--port` cmd line arg is renamed to `--frontend-port`

### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?

fixes #1064

### CODE REVIEW

File: pynecone/config.py [hunk 1]
```diff
@@ -198,6 +207,33 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
+        # set overriden class attribute values as os env variables to avoid losing them
+        for key, value in dict(self).items():
+            key = key.upper() if key != "port" else "FRONTEND_PORT"
```
Comment (picklelo) [inline pynecone/config.py:212]: Maybe we can rename the `--port` flag to `--frontend-port` to minimize confusion / having special cases. Would be a breaking change though

File: pynecone/constants.py [hunk 2]
```diff
@@ -4,9 +4,32 @@
 import re
 from enum import Enum
 from types import SimpleNamespace
+from typing import Type
 
 import pkg_resources
 
+
+def get_value(key, default=None, type_: Type = str) -> Type:
```
Comment (picklelo) [inline pynecone/constants.py:12]: Can we add type annotations for `key` and `default`

File: pynecone/constants.py [hunk 3]
```diff
@@ -4,9 +4,32 @@
...
+    """Get the value for the constant.
+    Obtain os env value and cast non-string types into
+    their original types.
+
+    Args:
+        key: constant name.
+        default: default value if key doesn't exist.
+        type_: the type of the constant
```
Comment (picklelo) [inline pynecone/constants.py:20]: nit: add newline between args/return

File: pynecone/constants.py [hunk 4]
```diff
@@ -306,5 +333,6 @@ class RouteRegex(SimpleNamespace):
 TOGGLE_COLOR_MODE = "toggleColorMode"
 
 # Server socket configuration variables
-CORS_ALLOWED_ORIGINS = "*"
+# CORS_ALLOWED_ORIGINS = "*"
```
Comment (picklelo) [inline pynecone/constants.py:336]: remove this?

File: pynecone/utils/build.py [hunk 5]
```diff
@@ -58,6 +58,14 @@ def set_environment_variables():
     )
 
 
+def set_os_env(**kwargs):
```
Comment (picklelo) [inline pynecone/utils/build.py:61]: Add docstrings for `kwargs`

**Review by picklelo — COMMENTED** [review]
Awesome work, just some minor comments and I think some updates to the unit tests to get the build passing

File: pynecone/config.py [hunk 6]
```diff
@@ -129,9 +132,9 @@ class Config(Base):
     username: Optional[str] = None
 
     # The frontend port.
-    port: str = constants.FRONTEND_PORT
+    frontend_port: str = constants.FRONTEND_PORT
```
Comment (picklelo) [inline pynecone/config.py:135]: Now I'm thinking maybe we can keep both `port` and `frontend_port` - and throw a deprecation warning on the old one. Just so we don't break anyone's app with the next release.

File: pynecone/utils/exec.py [hunk 7]
```diff
@@ -96,7 +96,7 @@ def run_frontend(
 
     # Run the frontend in development mode.
     console.rule("[bold green]App Running")
-    os.environ["PORT"] = get_config().port if port is None else port
+    os.environ["PORT"] = get_config().frontend_port if port is None else port
```
Comment (picklelo) [inline pynecone/utils/exec.py:99]: We should also make sure this matches the config

**Review by picklelo — APPROVED** [review]
Looks good! We can merge this in and do a follow up for the comments

**Comment by milochen0418** [comment]
Thanks a lot for this nice feature!

**Pull request merged**